src/include/duckdb/storage/table/column_data.hpp

```diff
@@ -167,8 +167,8 @@ class ColumnData : public enable_shared_from_this<ColumnData> {
 	//! Fetch the vector from the column data that belongs to this specific row
 	virtual idx_t Fetch(ColumnScanState &state, row_t row_id, Vector &result);
 	//! Fetch a specific row id and append it to the vector
-	virtual void FetchRow(TransactionData transaction, ColumnFetchState &state, row_t row_id, Vector &result,
-	                      idx_t result_idx);
+	virtual void FetchRow(TransactionData transaction, ColumnFetchState &state, const StorageIndex &storage_index,
+	                      row_t row_id, Vector &result, idx_t result_idx);
```

Comment:
This makes me wonder if `Select` and `Filter` also need to be dealt with in a similar fashion?
Or perhaps we're dodging that bullet by disabling pushdown extract when the filters on the LogicalGet contain struct extracts ?